awsbestpractices/missing-pagination/MissingPagination.java

```diff
@@ -0,0 +1,507 @@
+            log.info("Deleting file : {} with bucketName : {}", s3ObjectSummary.getKey(), bucketName);
+            keys.add(new DeleteObjectsRequest.KeyVersion(s3ObjectSummary.getKey()));
+        }
+        if (!keys.isEmpty()) {
+            s3Client.deleteObjects(new DeleteObjectsRequest(bucketName).withKeys(keys));
+        }
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/AWSServerlessAppsRepoStorage/blobs/6c58cdb1b0bffb657cea856afb1f450ef10a6003/--/src/com/amazonaws/serverlessappsrepo/storage/S3StorageService.java#L323-L341
+    public Set<String> missingPagination_3(final AmazonS3 s3Client, final String bucketName, final String prefix) {
+        ListObjectsV2Request listObjectsV2Request = new ListObjectsV2Request()
+                .withBucketName(bucketName)
+                .withPrefix(prefix);
+
+        ListObjectsV2Result listObjectsV2Result;
+
+        try (Timer.Context s3ListObjectsTimer = MetricBuilder.of(getClass(), "s3ListObjectsAPI.totalTime")
+                .timer()
+                .time()) {
+            log.info("List s3 objects.. bucket: {}, prefix: {}", bucketName, prefix);
+            listObjectsV2Result = s3Client.listObjectsV2(listObjectsV2Request);
+        }
+
+
+        return listObjectsV2Result.getObjectSummaries().stream()
+                .map(S3ObjectSummary::getKey)
+                .collect(Collectors.toList());
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/BrowseItemClassificationServiceLambdaJava/blobs/5eaf5c68ba4188fe786074cc1921f2af04d00be5/--/src/com/amazon/catalog/browse/bics/lambda/handler/MetricsQueryResultHandler.java#L41
+    public Set<String> missingPagination_4(Event event, Context context) throws InvalidRequestException {
+
+        checkNotNull(event, "Event cannot be null.");
+        checkNotNull(event.getExecutionContext(), "Event execution context cannot be null.");
+
+        final String executionContext = event.getExecutionContext();
+        if (executionContext.equals(EXECUTION_CONTEXT_CAC_ONLY)
+                || executionContext.equals(EXECUTION_CONTEXT_BOTH)) {
+
+            if (event.getPluginConfig() != null && event.getPluginConfig().containsKey(QUERY_EXECUTION_ID)) {
+
+                // Get Athena client
+                AmazonAthena client = null;
+                try {
+                    client = AthenaClientProvider.getAmazonAthena();
+                } catch (Exception e) {
+                    context.getLogger().log("Unexpected exception" + e);
+                    throw new IllegalStateException(e);
+                }
+
+                // Get the query result set
+                GetQueryResultsRequest queryResultsRequest = new GetQueryResultsRequest().withQueryExecutionId(event.getPluginConfig().get
+                        (QUERY_EXECUTION_ID));
+                GetQueryResultsResult queryResultsResult = client.getQueryResults(queryResultsRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.